piptools/scripts/compile.py

```diff
@@ -282,6 +282,8 @@ def cli(
         key_from_req(install_req.req): install_req for install_req in upgrade_reqs_gen
     }
 
+    existing_pins_to_upgrade = {}
```

Comment:
```suggestion
    existing_pins_to_upgrade = set()
```

Since the values ​​aren't used a `set ()` could be more suitable. What do you think?